project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/snapshot/ProjectCloningSnapshotManager.kt

```diff
@@ -72,13 +72,18 @@ class ProjectCloningSnapshotManager(rootProject: Project) : SnapshotManager {
         closeProject(context)
     }.onLeft { it.log() }
 
+    // TODO: JBRes-2103 Resource Management
     private suspend fun closeProject(context: IJDDContext) {
-        withContext(NonCancellable) {
+        withContext<Unit>(NonCancellable) {
             if (context is HeavyIJDDContext) {
                 ProjectManagerEx.getInstanceEx().forceCloseProjectAsync(context.project)
-            }
-            writeAction {
-                context.projectDir.run { delete(fileSystem) }
+            } else {
+                // avoid unnecessary project deletion, bad design (poebat' for now)
+                if (context.projectDir != context.indexProjectDir) {
+                    writeAction {
+                        context.projectDir.run { delete(fileSystem) }
+                    }
+                }
             }
         }
```

Comment:
Want we to have a pair of implementations for open/closing projects as `free`/`delete`? I mean move it to `ProjectOpeningService`

Reply:
That's also a good suggestion, but it's pretty specific for this snapshot manager. Moreover, we should have better resource management, especially in case of some faults. There is a ticket for this task [JBRes-2103](https://youtrack.jetbrains.com/issue/JBRes-2103)